modules/javafx.controls/src/test/java/test/javafx/scene/control/ComboBoxTest.java

```diff
@@ -1312,6 +1312,30 @@ public void defaultConverterCanHandleIncorrectType_2() {
         assertFalse(customCell.getPseudoClassStates().contains(empty));
     }
 
+    @Test
+    public void testPromptTextRestoredAfterSetValueNull() {
+        comboBox.setPromptText("Select Value");
+        comboBox.getItems().addAll("0", "1", "2", "3");
+
+        SingleSelectionModel<String> sm = comboBox.getSelectionModel();
+        sl = new StageLoader(comboBox);
+        comboBox.applyCss();
+        comboBox.show();
+        ListCell<String> buttonCell = (ListCell<String>) getDisplayNode();
+
+        assertEquals("Select Value", buttonCell.getText(), "Initial button cell text should be the promptText");
+
+        // Select an item
+        sm.select(2);
+        Toolkit.getToolkit().firePulse();
+
```

Comment:
An assert that the `buttonCell` is not `Select Value` would make sense here (`assertNotEquals`)